src/sources/file.rs

```diff
@@ -127,13 +130,16 @@ pub struct FileConfig {
     /// By default, the [global `log_schema.host_key` option][global_host_key] is used.
     ///
     /// [global_host_key]: https://vector.dev/docs/reference/configuration/global-options/#log_schema.host_key
-    #[serde(default)]
-    pub host_key: Option<OptionalValuePath>,
+    #[serde(default = "default_host_key")]
+    #[configurable(metadata(docs::examples = "hostname"))]
+    pub host_key: OptionalValuePath,
```

Comment:
I'm not sure making this not an `Option` works. It means the default value will be specified as the value of  `log_schema.host_key` when the docs are generated - which could be different to the value when Vector is actually run.

Reply:
We did the same in [internal_logs](https://github.com/vectordotdev/vector/blob/master/src/sources/internal_logs.rs#L35-L36). The existing docs are also hardcoded to display `host` ([ref](https://vector.dev/docs/reference/configuration/sources/file/#host_key)).

It's not actually an optional value, we do fill it in with the default if not provided - we just now do that as part of `serde` rather than in the code.

Reply:
I think it works? Might be missing something.
We / I have done this same thing on merged PRs so if it is incorrect I'll need to circle back and fix that.

Reply:
Doesn't that mean the internal_logs is wrong too?

You could configure:

```toml
log_schema.host_key = "groovyboogy"
```

and then not set the host_key here. The default value for `host_key` would then be `groovyboogy` and not `host` as the docs are now suggesting it would be.

Reply:
🤔 I think this is (and the PRs I did this in) a case where the current docs saying the default is "host" is wrong, because the default is actually whatever global log schema host key is.

Reply:
> Doesn't that mean the internal_logs is wrong too?

Sure, that would be an angle - but that also requires a change to every component that uses the log schema in the documentation. Using the `log_schema()` in the documentation at least keeps the default up-to-date with the default we've provided in the global configuration.

At worst it's no more incorrect than what was provided by the documentation before.